### PR TITLE
Th spies documentid

### LINKED ISSUE #44: Develop framework for spies testing for document.querySelectors
* Create framework for testing queryslectors through chai-spies

### LINKED ISSUE #46: Create Spies Testing for all querySelectors
* Use already created framework for querySelector spies testing to implement spies testing for all query selectors

### LINKED ISSUE #55: Fix bug where days exceed step goal method is returning too many days

### PR DESCRIPTION
**Pull request by relyt4me**
### What’s this PR do?
Completes the spies testing for getElementById used in the domUpdate portion. Minor changes made to sleep and scripts as cleanup while doing test.

### Where should the reviewer start?
Majority of changes to dom tests

### How should this be manually tested?
run npm test

### Any background context you want to provide?
Spies testing for entire graph may be too much data in a test file.

### What are the relevant tickets?
resolves #55 
resolves #46 
resolves #44

### CODE REVIEW

File: test/DomUpdates-test.js [hunk 1]
```diff
@@ -1,16 +1,62 @@
...
+  });
 
   beforeEach(() => {
     domUpdates = new DomUpdates();
-    chai.spy.on(domUpdates, ['populateUserWidget'], () => {});
+    global.document = {};
+    chai.spy.on(document, ['getElementById'], () => {
+      return { innerText: '' };
+    });
   });
```
Comment (jordy1611) [inline test/DomUpdates-test.js:61]: These are actually pretty thorough tests! Good job!

File: test/DomUpdates-test.js [hunk 2]
```diff
@@ -21,8 +67,55 @@ describe('DomUpdates', () => {
...
+    domUpdates.populateFriendsCard(currentUser, userRepo);
+
+    expect(document.getElementById).to.have.been.called(1);
+    expect(document.getElementById).to.have.been.called.with('friend-list');
+  });
+
+  it('Should call call getElementById each time changeInnerTextID is called with populateLeaderBoard', () => {
+    domUpdates.populateLeaderBoard(userRepo, sleepRepo, activityRepo, mostRecentDate, sleepRepo.sleepData);
+
+    expect(document.getElementById).to.have.been.called(7);
```
Comment (jordy1611) [inline test/DomUpdates-test.js:119]: If we have time should we have a test for anytime the getElementbyID method is called across all files?

File: test/DomUpdates-test.js (hunk 1 above)
Comment (carlymclift) [reply test/DomUpdates-test.js:61]: yeah you did a great job on this!

File: test/DomUpdates-test.js (hunk 2 above)
Comment (carlymclift) [reply test/DomUpdates-test.js:119]: yeah I think if we have time that would be a great idea!

**Pull request merged**